crates/sui-config/src/node.rs

```diff
@@ -330,6 +336,10 @@ impl ExpensiveSafetyCheckConfig {
     pub fn enable_move_vm_paranoid_checks(&self) -> bool {
         self.enable_move_vm_paranoid_checks
     }
+
+    pub fn enable_deep_per_tx_sui_conservation_check(&self) -> bool {
+        self.enable_deep_per_tx_sui_conservation_check
```

Comment:
Maybe check cfg!(debug_assertions) here instead?

Reply:
Good idea, and used this to simplify the code in `execution_engine` as well